### DISCUSSION TITLE
Открытие цикла в новой вкладке без перехода?

### DISCUSSION THREAD

**Discussion by erpic-ru** [totumonline/totum-ru-issues-and-discussions#134] [Как это сделать?] [answered]
Как можно реализовать открытие нескольких записей из таблицы циклов в отдельных вкладках бразуера без перехода?

У меня получилось реализовать такую задачу с помощью поля строка с кодом: = : str`#id + "/39"` и настройкой отображать как url в новой вкладке. 
<img width="558" alt="Снимок экрана 2022-05-18 в 10 27 09" src="https://user-images.githubusercontent.com/91674114/168982097-954f3832-a7bd-42e7-9f24-686fdf142d45.png">

Собственно, вопрос. Можно ли как-то заменить текст ссылки на символ или хотя бы дробную часть скрыть?

После подмены через форматирование перестает работать.

**Comment by ttm-alexeypolunin** [comment]
Вот код кнопки открытия цикла:

```
= : linkToTable(table: $table; cycle: #id; target: 'self' )
table: select(table: 'tables';  field: 'id' ; where: 'type'="calcs"; where: 'tree_node_id'=$nt; order: 'sort' )
nt: nowTableId()
```

Мы его копируем и немного модернизируем:
```
= : linkToTable(table: $table; cycle: #id; target: "blank")

		table: select(table: 'tables';  field: 'id' ; where: 'type'= "calcs"; where: 'tree_node_id'= $#nti; order: 'sort')
```

Теперь можно сделать еще кнопку с ним или сделать кнопку в контекстной панели, или сделать экшн по двойному клику для заблокированного поля и его уже оформить как `url`.

Можно и код основной кнопки `button_to_cycle` модифицировать — тогда в принципе будет открываться в новой вкладке.

**Reply by erpic-ru** [reply]
Так делать я умею. Дело в другом. Чтобы по клику на колесико цикл открывался в отдельной вкладке **без перехода в нее**, чтобы пользователь при этом остался в таблице циклов

**Reply by ttm-alexeypolunin** [reply]
А браузер вообще умеет так делать?

Я спрошу сег у разработки, если умеет, то мы добавим тип открытия для link-функций

**Reply by erpic-ru** [reply]
Да, мой вариант работает с заблокированной строкой. По колесику открывает и не переходит. Но само решение кустарное.

**Reply by ttm-alexeypolunin** [reply]
Попробуй сделать строковое поле с каким-нибудь содержимым и сделать в нем код действия как описано выше и включить [Execute action code on click (CodeActionOnClick)](https://docs.totum.online/fields#fields-settings-CodeActionOnClick) и [View the click as a URL (CodeActionOnClickAsUrl)](https://docs.totum.online/fields#fields-settings-CodeActionOnClick)

**Reply by erpic-ru** [reply]
Увы, оно тоже не работает. Ладно, просто я думал, может быстрое решение есть. Не сильно важная задача

**Reply by ttm-alexeypolunin** [reply]
В общем эта обработка именно ссылки и именно браузерная. В html нельзя сказать «вкладку в фоне».

Можно сформировать ссылку при помощи [panelHtml](https://docs.totum.online/functions#fn-panelHtml) к стандартной кнопке например.

**Comment by erpic-ru** [comment]
А в дальнейшем, можно куда-нибудь в бэклог закинуть задачу, чтобы в строковом поле при включенном параметре "отображать как ссылку" действие форматирования setformat ("text") не замещало адреса ссылки в коде, а только изменяло отображение значения в поле? 

Это даст возможность красиво размещать ссылки куда угодно в таблицах

**Reply by ttm-alexeypolunin** [reply]
Да, надо сделать

**Reply by ttm-alexeypolunin** [reply]
**готово в 3.4.43.0**

**Reply by erpic-ru** [reply]
Отчаянно применяю везде. Еще раз спасибо большое!
<img width="921" alt="Снимок экрана 2022-07-25 в 11 47 03" src="https://user-images.githubusercontent.com/91674114/180736806-b8f93e15-52d1-4be6-85db-d01c33c4de2f.png">
<br>
<img width="593" alt="Снимок экрана 2022-07-25 в 11 48 13" src="https://user-images.githubusercontent.com/91674114/180736923-26ef7eb8-08f0-4dd3-9e02-d1105f2e81d0.png">

**Comment by ttm-alexeypolunin** [comment, marked as answer]
Решил обновить инструкцию:

> Зачем это все — на кнопки не действует клик на 3 кнопку мыши и не получается открыть в новой вкладке инструментами браузера.

1. Делаем поле в таблице Циклов тип Строка в него суем ссылку на таблицу внутри цикла

2. Это можно сделать в аналогичном поле внутри цикла при создании цикла (код действия при добавлении) через `set` в таблицу циклов — функция `tableUrl` https://ru.docs.totum.online/functions#fn-tableUrl возврящает путь к таблице.

3. Потом мы делаем этому полю настройку `Открывать как url`

4. И через форматирование задаем ему читаемый `text` вместо ссылки

5. Теперь это ссылка и на нее действует третья кнопка мыши как открытие в новом окне.
